shell/core/plugins.js

```diff
@@ -88,8 +95,6 @@ export default function({
 
         removed.then(() => {
```

Comment:
If we used async/await in this function I think this code could be made more clear. As is we have everything in removed.then(()->...) being executed even if nothing has been removed.